src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFakeImpl.java

```diff
@@ -31,9 +33,25 @@ public void deleteUrlAlias(String email, String alias) {
     throw new UnsupportedOperationException();
   }
 
+  /**
+   * Create a list of UrlAlias objects by email.
+   *
+   * @param userEmail gets user email
+   * @return list of UrlAlias objects
+   * @throws edu.kpi.testcourse.storage.UrlRepository.PermissionDenied
```

Comment:
According to the [UrlRepository interface](https://github.com/future-stardust/url-shrtnr-palevo/blob/843bac6cdfc1942f04a9ed8e34a688e48496584e/src/main/java/edu/kpi/testcourse/storage/UrlRepository.java#L35), you must throw no error in this method!